### ISSUE TITLE
Remove Instances of Mock Accumulo 

### ISSUE THREAD

**Issue by james010101101** [gchq/Gaffer#1836]
In Accumulo 2.0.0 the mock accumulo store will be removed, in order to prepare for a later 2.0.0 migration we should remove all instances of the Mock Accumulo Store and replace them with Mini Accumulo Clusters

**Comment by james010101101** [comment]
The overheads of running Mini Accumulo raise the time taken for tests significantly. 
Integration tests in particular are a problem, refactoring the IT's to (where possible) use a single store/graph would be best.

**Comment by ctas582** [comment]
This hopefully shouldn't be that difficult. We are already running the integration tests against a single store when we run them on AWS. All we change is the store class, see: https://github.com/gchq/gaffer-tools/blob/master/deployment/aws/tests/integration-tests/emr-step-scripts/run-gaffer-integration-tests.sh#L118

**Comment by phosphorbronze50** [comment]
I have started to look into this issue.
Creating a MiniAccumuloStore and using this as the gaffer.store.class property has integrated moderately well into the testing regime but needs to be further enhanced to avoid resource issues of the Mini Accumulo Clusters not being gracefully closed after tests are complete. This will be tied into the refactoring of tests for performance issues already noted above.

**Comment by phosphorbronze50** [comment]
I have moved ahead with getting the MiniAccumuloCluster to start in new JUnit @BeforeClass / @BeforeAll methods and stop and tidy up in a new @AfterClass / @AfterAll methods.
The tests that simply specify the properties and supply them to the graph generation have been amended to read the store.properties:gaffer.store.class and instantiate the store if required.
The code that I am adding to the tests is non-Accumulo specific, using Store and StoreProperties classes, generating the Accumulo specifics.
The new Accumulo functionality has been added to the MiniAccumuloStore class itself.
In the MiniAccumuloStore many of the methods which get overridden in MockAccumuloStore now use the methods in the AccumuloStore superclass.
 
I would like to add a couple of additional methods to the Store Class :

public StoreProperties setUpTestDB(final StoreProperties properties) throws StoreException {
    return properties;
}

public void tearDownTestDB() {
    return;
}

These are concrete methods. 
They would be overridden in MiniAccumuloStore to call the necessary setup/start and stop/delete of the MiniAccumuloCluster. These overrides are then able to be called from the test classes' @BeforeClass / @BeforeAll and @AfterClass / @AfterAll methods.
MiniAccumuloStore.setUpTestDB() returns a clone of the StoreProperties that has the zookeeper and instance details updated to the values the MiniAccumuloCluster is using.
In the event of the class being run with a store.properties:gaffer.store.class property of AccumuloStore these methods effectively change nothing in the properties so all runs as it would have before.
Substituting a MiniAccumuloStore or SingleUse MiniAccumuloStore into the "store.properties:gaffer.store.class" has the updated JUnit tests working exactly as they did before. 
Putting the additional methods as "concrete placeholders" on the Store class means that the above @BeforeClass/@AfterClass mechanism is available to other, non-Accumulo, stores should that be needed in the future, but without requiring any changes to other stores at this time.

Is it OK to make these changes to the Store Class?

**Comment by phosphorbronze50** [comment]
I encountered issues in Mini Accumulo Cluster with users having bad authorizations and table permission exceptions.
This is because MockAccumulo we are moving from has no real concept of users.
I have corrected this as follows:
Create the MiniAccumuloCluster setting the root password etc.
Create the username listed in the store.properties with the password from the same place.
The user is given table create permission.
The user is given the Authorizations listed in the https://github.com/gchq/Gaffer/blob/develop/store-implementation/accumulo-store/README.md#tests
(vis1, vis2, public, private, publicVisibility, privateVisibility).
This seems to have enabled the MiniAccumuloStore to largely drop into where the MiniAccumuloStore gets used with no further issues.

**Comment by phosphorbronze50** [comment]
In uk.gov.gchq.gaffer.accumulostore.utils.TableUtilsTest
The test methods:
shouldThrowExceptionIfTableNameIsNotSpecified()
shouldThrowExceptionIfTableNameIsNotSpecifiedWhenCreatingTable()
have never worked properly, by the look of it.
They were annotated with @Test(expected = IllegalArgumentException.class).
This exception was thown when the store.initialise(null, schema, properties) was executed.
However, from what I can tell from the test method name the exception, and therefore a successful test, should have been triggered by the execution of TableUtils.ensureTableExists(store) and TableUtils.createTable(store), both of which should have actually thrown an AccumuloRuntimeException.
As far as I can see, and I could be wrong on this, there is actually no way to test the null table name scenarios because the store.initialise() will always fail with an IllegalArgumentException if the store has no/null table name.
In light of the above I will remove these two test methods.

**Comment by phosphorbronze50** [comment]
In uk.gov.gchq.gaffer.integration.graph.SchemaHidingIT I've had to make a small change as the superclass constructor accepted a file path reference to the store properties and read them in.
I have changed this to add an additional constructor that accepts a StoreProperties object and this then gets kept and used.
Because the MiniAccumuloStore implementation's properties requires the zookeeper etc amended and not wanting to write or amend a properties file into the codebase it made more sense to keep the properties as an object and just update what is necessary so the Graph creation happens in the temporary MiniAccumuloCluster.
The existing constructor is retained and all previous functionality works as before (I have tested the other subclasses that extend SchemaHidingIT and all are OK).
